lib/ansible/plugins/callback/telegraf.py

```diff
@@ -0,0 +1,290 @@
+            self._metric_prefix = os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_METRIC_PREFIX', None)
+        self._metric_summary = params.get('metric_summary', False)
+        if os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_METRIC_SUMMARY', None):
+            self._metric_summary = os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_METRIC_SUMMARY', None)
+
+        if self._display.verbosity > 1:
+            print('Using wire protocol {}, host {}, port {}'.format(wire_protocol, host, port))
+        if wire_protocol.startswith('http'):
+            try:
+                from telegraf import HttpClient
+                self.client = HttpClient(host=host, port=port)
+            except ImportError:
+                print('WARNING: Could not import telegraf HttpClient please pip install pytelegraf[http].  Using udp')
+                self.client = TelegrafClient(host=host, port=port)                
+        else:
+            try:
+                from telegraf.client import TelegrafClient
+                self.client = TelegrafClient(host=host, port=port)
+            except ImportError as e:
+                print('Could not import telegraf.client please pip install pytelegraf')
+                raise e
+
+        self._playbook_name = None
+
+        self.start_time = datetime.now()        
+        self._options = None
+        if cli:
+            self._options = cli.options
+
+        # self.playbook is either set by Ansible (v1), or by us in the `playbook_start` callback method (v2)
+        self.playbook = None
+        # self.play is either set by Ansible (v1), or by us in the `playbook_on_play_start` callback method (v2)
+        self.play = None
+
+    # Load parameters from conf file
+    def _load_conf(self, file_path):
+        conf_dict = {}
+        if file_path:
+            with open(file_path, 'r') as conf_file:
+                conf_dict = yaml.load(conf_file)
+
+        return conf_dict
+
+    # Send ansible metric to Telegraf
+    def send_metric(self, metric, value, host=None, tags={}):
+        """
+        @metric: '.' separated string metric name
+        @value: int, float or string
+        @tags: dict of point tags
+        @host: If set, over-ride source of metric (wavefront)
+        """
+        # Records a single value with one tag
+        # self.client.metric('some_metric', 123, tags={'server_name': 'my-server'})
+        merged_tags = self.default_tags
+        merged_tags.update(tags)
+        if host:
+            merged_tags['hostname'] = host
+
+        if self._metric_prefix:
+            metric = ".".join([self._metric_prefix, metric])
+        try:
+            self.client.metric(
+                metric,
+                value,
+                tags=merged_tags,
+            )
+        except Exception, e:
+            # We don't want Ansible to fail on an API error
+            print('Couldn\'t send metric "{0}" to telegraf'.format(metric))
```

Comment:
don't use print, display.error or raise an AnsibleError exception, Ansible already does not fail due to an Exception from a callback event.

Reply:
resolved